backend/src/main/java/org/cryptomator/hub/filters/VaultAdminOnlyFilterProvider.java

```diff
@@ -28,23 +33,44 @@ public class VaultAdminOnlyFilterProvider implements ContainerRequestFilter {
 	@Override
 	public void filter(ContainerRequestContext containerRequestContext) {
 		var vaultIdQueryParameter = getVaultIdQueryParameter(containerRequestContext);
-		var clientJwt = getClientJWT(containerRequestContext);
-		var unveridifedVaultId = getUnverifiedVaultId(clientJwt);
+		var clientJWT = getClientJWT(containerRequestContext);
+		var unveridifedVaultId = getUnverifiedVaultId(clientJWT);
 		if (vaultIdQueryParameter.equals(unveridifedVaultId)) {
 			var vault = Vault.<Vault>findByIdOptional(unveridifedVaultId).orElseThrow(NotFoundException::new);
 			var algorithm = Algorithm.ECDSA384(decodePublicKey(vault.authenticationPublicKey), null);
-			try {
-				JWT.require(algorithm).build().verify(clientJwt);
-			} catch (TokenExpiredException e) {
-				throw new VaultAdminTokenExpiredException("Token of Client-JWT expired");
-			} catch (JWTVerificationException e) {
-				throw new VaultAdminValidationFailedException("Different key used to sign the Client-JWT");
-			}
+			verify(buildVerification(algorithm), clientJWT);
 		} else {
 			throw new VaultAdminValidationFailedException("Other vaultId provided");
 		}
 	}
 
+	//visible for testing
+	void verify(JWTVerifier verifier, DecodedJWT clientJWT) {
+		try {
+			verifier.verify(clientJWT);
+		} catch (TokenExpiredException e) {
+			throw new VaultAdminTokenExpiredException("Token of Client-JWT expired");
+		} catch (IncorrectClaimException e) {
+			if (e.getClaimName().equals(RegisteredClaims.ISSUED_AT) || e.getClaimName().equals(RegisteredClaims.NOT_BEFORE)) {
+				throw new VaultAdminTokenNotYetValidException("Token of Client-JWT not yet valid");
+			} else {
+				throw new VaultAdminValidationFailedException("Incorrect claim exception");
+			}
+		} catch (JWTVerificationException e) {
+			throw new VaultAdminValidationFailedException("Different key used to sign the Client-JWT");
+		}
+	}
+
+	//visible for testing
+	JWTVerifier buildVerification(Algorithm algorithm) {
+		return buildVerifier(algorithm).build();
+	}
+
+	//visible for testing
+	Verification buildVerifier(Algorithm algorithm) {
+		return JWT.require(algorithm);
+	}
```

Comment:
1. do we really need both of these methods visible for testing separately?
2. why does `buildVerification` return a `verifier` and `buildVerifier` return a `verification`?
3. as noted in a different comment, I would suggest adding further verifications using the library's API instead of manually checking if fields are non-null:

```suggestion
	//visible for testing
	JWTVerifier buildVerifier(Algorithm algorithm) {
		return JWT.require(algorithm) //
			.withClaim("iat", (claim, jwt) -> jwt.getIssuedAt() != null) //
			.withClaim("nbf", (claim, jwt) -> jwt.getNotBefore() != null) //
			.withClaim("exp", (claim, jwt) -> jwt.getExpiresAt() != null) //
			.acceptLeeway(60) //
			.build();
	}
```

Reply:
1. Yes, because we can then use in tests the following
   ```
    private static final Clock NOW = Clock.fixed(Instant.ofEpochSecond(1516239020), ZoneId.of("UTC"));
    Mockito.doAnswer(invocationOnMock -> {
     Algorithm algorithm = invocationOnMock.getArgument(0);
     var verifier = (JWTVerifier.BaseVerification) vaultAdminOnlyFilterProvider.buildVerifier(algorithm);
     return verifier.build(NOW);
   }).when(vaultAdminOnlyFilterProvider).buildVerification(Mockito.any());
   ```
   which actually calls the real verifyer but only adds a fixed time to use static JWTs. This is only possible when we cast the `JWTVerifier` to `JWTVerifier.BaseVerification`. Otherwise we need to generate those JWTs in the test setup but I preferred a static setup with a modified time.
2. See 1.
3. I like it :+1: